src/WebJobs.Extensions.OpenAI/Embeddings/EmbeddingsHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.AI.OpenAI;
+
+namespace Microsoft.Azure.WebJobs.Extensions.OpenAI.Embeddings;
+static class EmbeddingsHelper
+{
+    static readonly char[] sentenceEndingsDefault = new[] { '.', '!', '?' };
+    static readonly char[] wordBreaksDefault = new[] { ',', ';', ':', ' ', '(', ')', '[', ']', '{', '}', '\t', '\n' };
+
+    public static EmbeddingsOptions BuildRequest(int maxOverlap, int maxChunkLength, string model, InputType inputType, string input)
+    {
+        using TextReader reader = GetTextReader(inputType, input);
+        if (maxOverlap >= maxChunkLength)
+        {
+            throw new ArgumentOutOfRangeException($"MaxOverlap ({maxOverlap}) must be less than MaxChunkLength ({maxChunkLength}).");
+        }
+
+        List<string> chunks = GetTextChunks(reader, 0, maxChunkLength, maxOverlap).ToList();
+        return new EmbeddingsOptions(model, chunks);
+    }
+
+    static TextReader GetTextReader(InputType inputType, string input)
+    {
+        if (inputType == InputType.RawText)
+        {
+            return new StringReader(input);
+        }
+        else if (inputType == InputType.FilePath)
+        {
+            return new StreamReader(input);
+        }
+        else if (inputType == InputType.URL)
+        {
+            HttpClient client = new();
+            Stream stream = client.GetStreamAsync(input).Result;
```

Comment:
Calling `.Result` on a task can lead to thread starvation or worse, deadlocks. We should instead use async/await.

Reply:
updated to use async await